### ISSUE TITLE
suggestion: feat. that switch crypto if too many results are rejected

### ISSUE THREAD

**Issue by krrkrr** [fireice-uk/xmr-stak#2364]
Hi...

I'm not sure it's the right place for my request, but I can't find another way....

With the Monero being "updated" on a regular basis, to improve resistance against ASIC, it might sometimes lead one to have several miners to mine after the chain update.... without miners being updated. Which is a waste of ressources....

So, i'd like to suggest a feature, that would work like that: when there's a certain amount of results rejected by a pool, (number amount that would be in a conf file, for instance, to allow user to set it), the miner will change the pool and/or the crypto mined. 

In this scenario, if I got some XMR miners that can't be updated on the next chain update, they will automatically switch to Ryo.. or XMO... or etc...

I hope I was clear enough. Thanks for the work, anyway !!!

**Comment by Spudz76** [comment]
This is more something for a wrapping controller like meta-miner
it is for automatic coin jumping for profit, but could be easily modified to blacklist broken miners
and to support more than one pool etc

but it works like a proxy and monitors whats coming in and out of your miner applet and pool, by default in 600 seconds it will kill and retry a miner that has not provided any good results
and works with whatever miner applet of the day

**Comment by krrkrr** [comment]
using Meta-miner... instead of xmr-stak ? or over xmr-stak ? does MetaMiner works on macOS ?

**Comment by psychocrypt** [comment]
If the miner works fine you should have no rejected shares (or only a few
because of the delay between job receive and submit)

If you have a high rate of rejected shares a coin switch will not help

**Comment by krrkrr** [comment]
ah ? So I can keep mining XMR with xmr-stak 2.5.0 ????

**Comment by psychocrypt** [comment]
no you need the latest version since monero fored to cryptonight_r

**Comment by krrkrr** [comment]
that's why I was asking for a feature that allows xmr-stak to switch automatically from XMR to another crypto, to avoid losing minig resources when there's such an update and one can't update all his miners for several days/weeks...

**Comment by psychocrypt** [comment]
But if the other currencies also update you will have the same problem.
Normally a fork is know at least 1 month before it will happen.
Do you have no remote access?

Sry but such a feature will be not included. There is no time to build it.
:(

**Comment by Spudz76** [comment]
yes meta-miner should work on OSX it's Node.JS so it runs wherever you can have that
it supervises miner processes underneath it by proxying their pool traffic (so it can see what it's doing) and tracking the PIDs so it can kill them off when they are wrong

wrong in the current case being unprofitable but this funky "by any coin necessary" sort of thing could be added (by you)
